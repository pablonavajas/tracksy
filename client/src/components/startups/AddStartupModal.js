import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addStartup } from '../../actions/startupsActions';

import M from 'materialize-css/dist/js/materialize.min.js';

const AddStartupModal = ({ addStartup }) => {
  const [name, setName] = useState('');
  const [website, setWebsite] = useState('');
  const [ownership, setOwnership] = useState('');
  const [board, setBoard] = useState('');
  const [investment_1, setInvestment_1] = useState('');
  const [type_1, setType_1] = useState('');
  const [date_1, setDate_1] = useState('');
  const [investment_2, setInvestment_2] = useState('');
  const [type_2, setType_2] = useState('');
  const [date_2, setDate_2] = useState('');

  const onSubmit = () => {
    if (
      name === '' ||
      website === '' ||
      ownership === '' ||
      board === '' ||
      investment_1 === '' ||
      type_1 === '' ||
      date_1 === '' ||
      investment_2 === '' ||
      type_2 === '' ||
      date_2 === ''
    ) {
      M.toast({
        html: 'Please enter NA in the fields that are not applicable'
      });
    } else {
      const newStartup = {
        name,
        website,
        ownership,
        board,
        investment_1,
        investment_2,
        type_1,
        type_2,
        date_1,
        date_2
      };

      addStartup(newStartup);

      // Clear fields
      setName('');
      setWebsite('');
      setOwnership('');
      setBoard('');
      setInvestment_1('');
      setDate_1('');
      setType_1('');
      setInvestment_2('');
      setDate_2('');
      setType_2('');
    }
    //console.log('Submit button Clicked');
  };

  return (
    <div id="add-startup-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Enter New Startup</h4>
        {/** Name */}
        <div className="row">
          <div className="input-field">
            <input
              id="name"
              type="text"
              name="name"
              value={name}
              onChange={e => setName(e.target.value)}
            />
            <label>Startup Name</label>
          </div>
        </div>

        {/** Website */}
        <div className="row">
          <div className="input-field">
            <input
              id="website"
              type="text"
              name="name"
              value={website}
              onChange={e => setWebsite(e.target.value)}
            />
            <label>Website</label>
          </div>
        </div>

        {/** Ownership */}
        <div className="row">
          <div className="input-field">
            <input
              id="ownership"
              name="ownership"
              type="text"
              value={ownership}
              onChange={e => setOwnership(e.target.value)}
            />
            <label>Ownership (%)</label>
          </div>
        </div>

        {/** Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="investment_1"
              name="investment_1"
              type="text"
              value={investment_1}
              onChange={e => setInvestment_1(e.target.value)}
            />
            <label>Investment 1</label>
          </div>
        </div>
        {/** Type of Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="type_1"
              name="type_1"
              type="text"
              value={type_1}
              onChange={e => setType_1(e.target.value)}
            />
            <label>Type of Investment 1</label>
          </div>
        </div>
        {/** Closing Date Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="date_1"
              name="date_1"
              value={date_1}
              type="date"
              onChange={e => setDate_1(e.target.value)}
            />
            <label>Close Date</label>
          </div>
        </div>

        {/** Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="investment_2"
              name="investment_2"
              type="text"
              value={investment_2}
              onChange={e => setInvestment_2(e.target.value)}
            />
            <label>Investment 2</label>
          </div>
        </div>

        {/** Type of Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="type_2"
              name="type_2"
              type="text"
              value={type_2}
              onChange={e => setType_2(e.target.value)}
            />
            <label>Type of Investment 2</label>
          </div>
        </div>

        {/** Closing Date Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="date_2"
              name="date_2"
              value={date_2}
              type="date"
              onChange={e => setDate_2(e.target.value)}
            />
            <label>Close</label>
          </div>
        </div>
      </div>
      <div className="row">
        <div className="input-field col s12">
          <select
            id="board"
            name="board"
            value={board}
            className="browser-default"
            onChange={e => setBoard(e.target.value)}
          >
            <option value="" disabled>
              Select Board
            </option>
            <option value="Member">Member</option>
            <option value=" - "> - </option>
            <option value="Observer"> Observer </option>
          </select>
        </div>
      </div>
      <div className="modal-footer">
        <a
          href="#!"
          onClick={onSubmit}
          className="modal-close waves-effect blue waves-light btn"
        >
          Submit
        </a>
      </div>
    </div>
  );
};

const modalStyle = {
  width: '60%',
  height: '75%'
};

AddStartupModal.propTypes = {
  addStartup: PropTypes.func.isRequired
};

export default connect(null, { addStartup })(AddStartupModal);
