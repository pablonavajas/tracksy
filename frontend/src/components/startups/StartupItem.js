import React from "react";
import PropTypes from "prop-types";
import CurrencyFormat from "react-currency-format";
var moment = require("moment");

import { connect } from "react-redux";
import { deleteStartup, setCurrent } from "../../actions/startupsActions";
import { ClickIcon } from "../smallComponents/displayComponents";
import { Chip } from "../smallComponents/displayComponents";

export const StartupItem = ({ startup, setCurrent, deleteStartup }) => {
  return (
    <tr>
      <th className="center">
        <a href={`https://${startup.website}`}>
          <img
            src={`http://logo.clearbit.com/${startup.website}`}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">
        <Chip data={startup.ownership} />
      </td>
      <td className="center">
        <Chip data={startup.board} />
      </td>
      <td className="center">
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip
              data={
                <CurrencyFormat
                  value={investment.value}
                  displayType={"text"}
                  thousandSeparator={true}
                  prefix={investment.currency}
                />
              }
            />
          </div>
        ))}
      </td>
      <td className="center">
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip data={investment.investmentType} />
          </div>
        ))}
      </td>
      <td>
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip data={moment(investment.date).format("DD/MM/YYYY")} />
          </div>
        ))}
      </td>
      <td>
        <ClickIcon
          href="#startup-overview"
          setFunction={setCurrent}
          val={startup}
          iconName="dehaze"
        />
      </td>
      <td>
        <ClickIcon
          href="#edit-startup-modal"
          setFunction={setCurrent}
          val={startup}
          iconName="edit"
        />
      </td>
      <td>
        <ClickIcon
          href="#add-investments-modal"
          setFunction={setCurrent}
          val={startup}
          iconName="attach_money"
        />
      </td>
      {/* Display grey button with no functionality if some financials have already been added */}
      {startup.financials.length === 0 ? (
        <td>
          <ClickIcon
            href="#add-kpi-names-modal"
            setFunction={setCurrent}
            val={startup}
            iconName="insert_chart"
          />
        </td>
      ) : (
        <td>
          <a
            id="unclickableIcon"
            href="!#"
            className="secondary-content"
            onClick={(e) => e.preventDefault()}
          >
            <i className="material-icons grey-text">insert_chart</i>
          </a>
        </td>
      )}
      <td>
        <a
          id="deleteIcon"
          href="#!"
          onClick={(e) => {
            e.preventDefault();
            deleteStartup(startup.id);
          }}
          className="secondary-content"
        >
          <i className="material-icons blue-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

StartupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired,
};

// Inject App Level state, so that
// component updates upon an update of info
// (e.g. added investment)
const mapStateToProps = (state) => ({
  startups: state.startup.startups,
});

export default connect(mapStateToProps, { setCurrent, deleteStartup })(
  StartupItem
);
