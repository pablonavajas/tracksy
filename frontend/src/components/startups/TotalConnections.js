import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import {getConnections} from  '../../actions/connectionsActions';


const TotalConnections = ({ current, connections, getConnections }) => {
    const [name, setName] = useState("");
    const [website, setWebsite] = useState("");

    useEffect(() => {
    getConnections();
  }, [connections, current]);

  if (connections === null) {
    return (
      <div className="progress">
        <div className="indeterminate"></div>
      </div>
    );
  }

  return (
  // Section: Stats
  
  <section className="section section-stats center">
    <nav>
    <div className="nav-wrapper blue-grey darken-4">
      <ul id="nav-mobile" className="left hide-on-med-and-down">
        <li><a href="#startup-overview">Overview</a></li>
        <li><a href="#hiring">Hiring</a></li>
        <li className="active"><a href="#total-connections">Connections</a></li>
      </ul>
    </div>
  </nav>
  
  <div className="col s12 m6 l4">
          <div className="card">
            <div className="card-content">
              <table className="striped">
                <thead>
                  <tr className = "center">
                    <th>Name</th>
                    <th>Description</th>
                    <th>URL</th>
                  </tr>
                </thead>
                <tbody>
                   {connections.map((c) => (
                     <tr key={c.id}>
                       <td> {c.name} </td>
                       <td> {c.description} </td>
                       <td> {c.url} </td>
                   </tr>
                   ))}
                </tbody>
              </table>
            </div>
        </div>
        </div>

      </section>
  );
};


TotalConnections.propTypes = {
  current: PropTypes.object,
  getConnections:PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  connections: state.connections,
  current: state.startup.current
});

export default connect(mapStateToProps, { getConnections })(TotalConnections);
