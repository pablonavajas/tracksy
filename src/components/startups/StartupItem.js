import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { deleteStartup, setCurrent } from '../../actions/startupsActions';

import M from 'materialize-css/dist/js/materialize.min.js';

const startupItem = ({ startup, setCurrent, deleteStartup }) => {
  const onDelete = () => {
    deleteStartup(startup.id);
    M.toast({ html: 'Startup Deleted' });
  };

  return (
    <tr>
      <th className="center">
        <a href={'https://' + startup.website}>
          <img
            src={'http://logo.clearbit.com/' + startup.website}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">{startup.ownership}</td>
      <td className="center">{startup.board}</td>
      <td>{startup.investment_1}</td>
      <td className="center">{startup.type_1}</td>
      <td>{startup.date_1}</td>
      <td className="center">{startup.investment_2}</td>
      <td className="center">{startup.type_2}</td>
      <td>{startup.date_2}</td>
      <td>
        <a
          href="#edit-startup-modal"
          onClick={() => setCurrent(startup)}
          className="secondary-content modal-trigger"
        >
          <i className="material-icons grey-text">edit</i>
        </a>
      </td>
      <td width="50">
        <a href="#!" onClick={onDelete} className="secondary-content">
          <i className="material-icons grey-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

startupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired
};

export default connect(null, { setCurrent, deleteStartup })(startupItem);
