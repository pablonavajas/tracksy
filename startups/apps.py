from django.apps import AppConfig


class TracksyappConfig(AppConfig):
    name = 'startups'
