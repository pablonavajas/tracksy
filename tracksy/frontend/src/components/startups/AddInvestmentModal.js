import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CurrencyFormat from "react-currency-format";

import {
  deleteInvestment,
  addInvestments,
} from "../../actions/investmentsActions";
import { setCurrent } from "../../actions/startupsActions";

const addInvestmentModal = ({
  current,
  addInvestments,
  deleteInvestment,
  setCurrent,
}) => {
  const [currentName, setCurrentName] = useState("");
  const [list, setList] = useState([]);

  useEffect(() => {
    if (current) {
      setCurrentName(current.name);
      setList(current.investments);
    }
  }, [current]);

  const initialInputState = {
    value: "",
    currency: "£",
    investmentType: "",
    date: "",
  };

  const addInput = () => {
    setList([...list, initialInputState]);
  };

  const onSubmit = () => {
    addInvestments(current.id, list);
    setCurrent(null);
  };

  // 0-based index
  const deleteInput = (index) => {
    // delete the inputs from database if they were already there
    if (list[index].id) {
      deleteInvestment(current.id, list[index].id);
    }

    //delete the inputs from the modal
    list.splice(index, 1);
    setList([...list]);
  };

  return (
    <div id="add-investments-modal" className="modal" style={modalStyle}>
      <div>
        <div className="modal-content">
          <h4>Investments for {currentName}</h4>
          <div className="center">
            {list.map((inv, i) => (
              <div className="row" key={i}>
                <div className="input-field col s1">
                  <select
                    id={"invCurrency" + i}
                    name="currency"
                    value={inv.currency ? inv.currency : ""}
                    className="browser-default"
                    onChange={(e) => {
                      list[i].currency = e.target.value;
                      setList([...list]);
                    }}
                  >
                    <option value="£" defaultValue="£">
                      £
                    </option>
                    <option value="$">$</option>
                    <option value="€">€</option>
                  </select>
                  <label htmlFor={"invCurrency" + i} className="active">
                    Currency
                  </label>
                </div>
                <div className="input-field col s4" key={i}>
                  <CurrencyFormat
                    id={"invValue" + i}
                    value={inv.value}
                    thousandSeparator={true}
                    className="validate"
                    prefix={inv.currency ? inv.currency : "£"}
                    allowNegative={false}
                    onValueChange={(values) => {
                      const { formattedValue, value } = values;
                      // formattedValue = $2,223
                      // value ie, 2223
                      list[i].value = value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"inveValue" + i} className="active">
                    Investment
                  </label>
                </div>
                <div className="input-field col s4">
                  <input
                    placeholder=""
                    id={"invType" + i}
                    name="type_1"
                    type="text"
                    className="validate"
                    value={inv.investmentType ? inv.investmentType : ""}
                    onChange={(e) => {
                      list[i].investmentType = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invType" + i} className="active">
                    Type of Investment
                  </label>
                </div>
                <div className="input-field col s2">
                  <input
                    id={"invDate" + i}
                    name="date_1"
                    value={inv.date ? inv.date : ""}
                    type="date"
                    className="validate"
                    onChange={(e) => {
                      list[i].date = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invDate" + i} className="active">
                    Close Date
                  </label>
                  <a href="#" onClick={() => deleteInput(i)}>
                    <i className="material-icons grey-text prefix">
                      delete_forever
                    </i>
                  </a>
                </div>
              </div>
            ))}

            <a
              className="btn waves-effect waves-light light-blue"
              onClick={addInput}
            >
              Add Investment
            </a>
          </div>
        </div>
        <div className="modal-footer">
          <div className="center">
            <a
              onClick={onSubmit}
              className="modal-close waves-effect light-blue waves-light btn"
            >
              Submit
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%",
};

addInvestmentModal.propTypes = {
  addInvestments: PropTypes.func.isRequired,
  deleteInvestment: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  current: state.startup.current,
});

export default connect(mapStateToProps, {
  addInvestments,
  deleteInvestment,
  setCurrent,
})(addInvestmentModal);
