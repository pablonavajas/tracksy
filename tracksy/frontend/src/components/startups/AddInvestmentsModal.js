import React, { useState, useEffect, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { addInvestment } from "../../actions/investmentsActions";
import CurrencyFormat from "react-currency-format";
import { setCurrent } from "../../actions/startupsActions";

const addInvestmentModal = ({ addInvestment, current }) => {
  const [currentName, setCurrentName] = useState("Hello");
  const [list, setList] = useState([
    {
      startupId: null,
      value: 0,
      currency: "£",
      investmentType: "",
      date: ""
    }
  ]);

  useEffect(() => {
    if (current) {
      setCurrentName(current.name);
      setList(current.investments);
    }
  }, [current]);

  const addInput = () => {
    list.push({
      startupId: current.id,
      value: "",
      currency: "£",
      investmentType: "",
      date: ""
    });
    setList([...list]);
  };

  const onSubmit = () => {
    list.map(inv => {
      addInvestment(inv);
    });

    // Clear fields
    // setList([]);
    setList([
      {
        startupId: null,
        value: 0,
        currency: "£",
        investmentType: "",
        date: ""
      }
    ]);
  };

  return (
    <div id="add-investments-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Enter Investments for {currentName}</h4>
        <div className="center">
          <div className="row">
            {list.map((inv, i) => (
              <div key={i}>
                <div className="input-field col s1">
                  <select
                    id={"invCurrency" + i}
                    name="currency"
                    value={inv.currency}
                    className="browser-default"
                    onChange={e => {
                      list[i].currency = e.target.value;
                      setList([...list]);
                    }}
                  >
                    <option value="£" defaultValue="£">
                      £
                    </option>
                    <option value="$">$</option>
                    <option value="€">€</option>
                  </select>
                  <label htmlFor={"invCurrency" + i} className="active">
                    Currency
                  </label>
                </div>
                <div className="input-field col s4" key={i}>
                  <CurrencyFormat
                    id={"invValue" + i}
                    value={inv.value}
                    thousandSeparator={true}
                    prefix={inv.currency}
                    allowNegative={false}
                    onValueChange={values => {
                      const { formattedValue, value } = values;
                      // formattedValue = $2,223
                      // value ie, 2223
                      list[i].startupId = current.id;
                      list[i].value = value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"inveValue" + i} className="active">
                    Investment
                  </label>
                </div>
                <div className="input-field col s4">
                  <input
                    placeholder=""
                    id={"invType" + i}
                    name="type_1"
                    type="text"
                    value={inv.investmentType}
                    onChange={e => {
                      list[i].investmentType = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invType" + i} className="active">
                    Type of Investment
                  </label>
                </div>
                <div className="input-field col s3">
                  <input
                    id={"invDate" + i}
                    name="date_1"
                    value={inv.date}
                    type="date"
                    onChange={e => {
                      list[i].date = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invDate" + i} className="active">
                    Close Date
                  </label>
                </div>
              </div>
            ))}
          </div>
          <a
            className="btn waves-effect waves-light light-blue"
            onClick={addInput}
          >
            Add Investment
          </a>
        </div>
      </div>
      <div className="modal-footer">
        <a
          onClick={onSubmit}
          className="modal-close waves-effect blue waves-light btn-flat"
        >
          Submit
        </a>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%"
};

addInvestmentModal.propTypes = {
  addInvestment: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  // console.log(state.startup.current);
  return {
    current: state.startup.current
  };
};

export default connect(mapStateToProps, { addInvestment })(addInvestmentModal);
