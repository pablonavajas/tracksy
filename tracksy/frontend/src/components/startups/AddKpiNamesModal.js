import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { addKpiNames, deleteKpiName } from "../../actions/kpiActions";
import { setCurrent } from "../../actions/startupsActions";

export const AddKpiNamesModal = ({
  current,
  addKpiNames,
  deleteKpiName,
  setCurrent,
}) => {
  const [currentName, setCurrentName] = useState("");
  const [defaultKpis, setDefaultKPIs] = useState([
    {
      name: "Revenue",
    },
    {
      name: "Cash Balance",
    },
    {
      name: "Monthly Burn",
    },
  ]);

  let kpiNo = 0;
  const incrementKpiNo = () => {
    kpiNo += 1;
  };
  const decrementKpiNo = () => {
    kpiNo -= 1;
  };

  const [list, setList] = useState([{ name: "" }]);

  useEffect(() => {
    if (current) {
      setCurrentName(current.name);
      setList(current.kpinames);
    }
  }, [current]);

  const addInput = () => {
    incrementKpiNo();
    setList([...list, { name: "" }]);
  };

  // 0-based index
  const deleteInput = (index) => {
    // delete the inputs from database if they were already there
    if (list[index].id) {
      deleteKpiName(current.id, list[index].id);
    }

    //delete the inputs from the modal
    decrementKpiNo();
    list.splice(index, 1);
    setList([...list]);
  };

  const onSubmit = () => {
    addKpiNames(current.id, list);
    setCurrent(null);
  };

  return (
    <div id="add-kpi-names-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Enter KPI Names for {currentName}</h4>
        <div className="center">
          <div className="row">
            {defaultKpis.map((kpi) => {
              incrementKpiNo();
              return (
                <div className="input-field col s4" key={kpiNo}>
                  <input disabled value={kpi.name} id={kpiNo} type="text" />
                  <label htmlFor={kpiNo}>Kpi Name {kpiNo} (default)</label>
                </div>
              );
            })}
            {list.map((kpi, i) => (
              <div key={i}>
                <div className="input-field col s4">
                  <input
                    placeholder=""
                    id={"kpiName" + i}
                    name="type_1"
                    type="text"
                    className="validate"
                    value={kpi.name}
                    onChange={(e) => {
                      list[i].name = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"kpiName" + i} className="active">
                    KPI Name {kpiNo + 1 + i}
                  </label>
                  <a
                    id="deleteKpiNameButton"
                    href="#"
                    onClick={() => deleteInput(i)}
                  >
                    <i className=" material-icons grey-text postfix">
                      delete_forever
                    </i>
                  </a>
                </div>
              </div>
            ))}
          </div>
          <a
            id="addKpiNameButton"
            className="btn waves-effect waves-light light-blue"
            onClick={addInput}
          >
            Add KPI Name
          </a>
        </div>
      </div>
      <div className="modal-footer">
        <div className="center">
          <a
            id="submitKpiNamesButton"
            onClick={onSubmit}
            className="modal-close waves-effect light-blue waves-light btn"
          >
            Submit
          </a>
        </div>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%",
};

AddKpiNamesModal.propTypes = {
  addKpiNames: PropTypes.func.isRequired,
  deleteKpiName: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  current: state.startup.current,
});

export default connect(mapStateToProps, {
  addKpiNames,
  deleteKpiName,
  setCurrent,
})(AddKpiNamesModal);
