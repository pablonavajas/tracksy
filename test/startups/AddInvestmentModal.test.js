import React from "react";
import { shallow, mount } from "enzyme";
import { AddInvestmentModal } from "../../tracksy/frontend/src/components/startups/AddInvestmentModal";
import ConnectedAddInvestmentModal from "../../tracksy/frontend/src/components/startups/AddInvestmentModal";
import toJson from "enzyme-to-json";

import "../setupTests";

import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
const mockStore = configureMockStore([thunk]);

//Snapshot Test
describe("AddInvestmentModal ", () => {
  let wrapper;
  const deleteInvestment = jest.fn();
  const addInvestments = jest.fn();
  const setCurrent = jest.fn();

  it("renders correctly with no investments", () => {
    wrapper = shallow(
      <AddInvestmentModal
        addInvestments={addInvestments}
        deleteInvestment={deleteInvestment}
        setCurrent={setCurrent}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders correctly with investments", () => {
    const current = {
      id: 0,
      investments: [
        {
          id: 0,
          value: 10,
          currency: "$",
          investmentType: "A",
          date: "10-10-2000",
        },
        {
          id: 1,
          value: 20,
          currency: "$",
          investmentType: "B",
          date: "10-10-2001",
        },
      ],
    };
    wrapper = mount(
      <AddInvestmentModal
        addInvestments={addInvestments}
        deleteInvestment={deleteInvestment}
        setCurrent={setCurrent}
        current={current}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("AddInvestmentModal ", () => {
  let wrapper;
  const deleteInvestment = jest.fn();
  const addInvestments = jest.fn();
  const setCurrent = jest.fn();
  const current = {
    id: 0,
    investments: [
      { id: 0, value: 10 },
      { id: 1, value: 20 },
    ],
  };

  beforeEach(() => {
    wrapper = mount(
      <AddInvestmentModal
        addInvestments={addInvestments}
        deleteInvestment={deleteInvestment}
        setCurrent={setCurrent}
        current={current}
      />
    );
    wrapper.update();
  });

  it("when submit is clicked, addInvestments is called and current is set to null", () => {
    wrapper.find("#submitInvestmentButton").simulate("click");
    expect(addInvestments).toHaveBeenCalledWith(0, current.investments);
    expect(setCurrent).toHaveBeenCalledWith(null);
  });

  it("when add investment button is clicked, internal list is updated", () => {
    // this button, uses React Hook UseState, therefore no need to test,
    // included for the purposes of coverage
    wrapper.find("#addInvestmentButton").simulate("click");
  });
  it("hehe", () => {
    wrapper.find("#deleteInvestmentButton").at(1).simulate("click");
    expect(deleteInvestment).toHaveBeenCalledWith(current.id, 1);
  });
});

// Testing MapStateToProps
describe("AddInvestmentModal ", () => {
  const deleteInvestment = jest.fn();
  const addInvestments = jest.fn();
  const setCurrent = jest.fn();
  const current = {
    id: 1,
    name: "test",
  };

  const initialState = {
    startup: { current },
  };

  const store = mockStore(initialState);
  const wrapper = shallow(
    <ConnectedAddInvestmentModal
      addInvestments={addInvestments}
      deleteInvestment={deleteInvestment}
      setCurrent={setCurrent}
      store={store}
    />
  );

  it("should show the current value imported from store", () => {
    expect(wrapper.props().children.props.current).toEqual(current);
  });
});
