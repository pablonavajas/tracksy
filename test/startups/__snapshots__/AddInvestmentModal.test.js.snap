// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddInvestmentModal  renders correctly with investments 1`] = `
<AddInvestmentModal
  addInvestments={[MockFunction]}
  current={
    Object {
      "id": 0,
      "investments": Array [
        Object {
          "currency": "$",
          "date": "10-10-2000",
          "id": 0,
          "investmentType": "A",
          "value": 10,
        },
        Object {
          "currency": "$",
          "date": "10-10-2001",
          "id": 1,
          "investmentType": "B",
          "value": 20,
        },
      ],
    }
  }
  deleteInvestment={[MockFunction]}
  setCurrent={[MockFunction]}
>
  <div
    className="modal"
    id="add-investments-modal"
    style={
      Object {
        "height": "75%",
        "width": "60%",
      }
    }
  >
    <div>
      <div
        className="modal-content"
      >
        <h4>
          Investments for 
        </h4>
        <div
          className="center"
        >
          <div
            className="row"
            key="0"
          >
            <div
              className="input-field col s1"
            >
              <select
                className="browser-default"
                id="invCurrency0"
                name="currency"
                onChange={[Function]}
                value="$"
              >
                <option
                  defaultValue="£"
                  value="£"
                >
                  £
                </option>
                <option
                  value="$"
                >
                  $
                </option>
                <option
                  value="€"
                >
                  €
                </option>
              </select>
              <label
                className="active"
                htmlFor="invCurrency0"
              >
                Currency
              </label>
            </div>
            <div
              className="input-field col s4"
              key="0"
            >
              <CurrencyFormat
                allowNegative={false}
                className="validate"
                decimalSeparator="."
                displayType="input"
                fixedDecimalScale={false}
                id="invValue0"
                isAllowed={[Function]}
                isNumericString={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseUp={[Function]}
                onValueChange={[Function]}
                prefix="$"
                suffix=""
                thousandSeparator={true}
                thousandSpacing="3"
                type="text"
                value={10}
              >
                <input
                  className="validate"
                  id="invValue0"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  type="text"
                  value="$10"
                />
              </CurrencyFormat>
              <label
                className="active"
                htmlFor="inveValue0"
              >
                Investment
              </label>
            </div>
            <div
              className="input-field col s4"
            >
              <input
                className="validate"
                id="invType0"
                name="type_1"
                onChange={[Function]}
                placeholder=""
                type="text"
                value="A"
              />
              <label
                className="active"
                htmlFor="invType0"
              >
                Type of Investment
              </label>
            </div>
            <div
              className="input-field col s2"
            >
              <input
                className="validate"
                id="invDate0"
                name="date_1"
                onChange={[Function]}
                type="date"
                value="10-10-2000"
              />
              <label
                className="active"
                htmlFor="invDate0"
              >
                Close Date
              </label>
              <a
                href="#"
                id="deleteInvestmentButton"
                onClick={[Function]}
              >
                <i
                  className="material-icons grey-text prefix"
                >
                  delete_forever
                </i>
              </a>
            </div>
          </div>
          <div
            className="row"
            key="1"
          >
            <div
              className="input-field col s1"
            >
              <select
                className="browser-default"
                id="invCurrency1"
                name="currency"
                onChange={[Function]}
                value="$"
              >
                <option
                  defaultValue="£"
                  value="£"
                >
                  £
                </option>
                <option
                  value="$"
                >
                  $
                </option>
                <option
                  value="€"
                >
                  €
                </option>
              </select>
              <label
                className="active"
                htmlFor="invCurrency1"
              >
                Currency
              </label>
            </div>
            <div
              className="input-field col s4"
              key="1"
            >
              <CurrencyFormat
                allowNegative={false}
                className="validate"
                decimalSeparator="."
                displayType="input"
                fixedDecimalScale={false}
                id="invValue1"
                isAllowed={[Function]}
                isNumericString={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseUp={[Function]}
                onValueChange={[Function]}
                prefix="$"
                suffix=""
                thousandSeparator={true}
                thousandSpacing="3"
                type="text"
                value={20}
              >
                <input
                  className="validate"
                  id="invValue1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  type="text"
                  value="$20"
                />
              </CurrencyFormat>
              <label
                className="active"
                htmlFor="inveValue1"
              >
                Investment
              </label>
            </div>
            <div
              className="input-field col s4"
            >
              <input
                className="validate"
                id="invType1"
                name="type_1"
                onChange={[Function]}
                placeholder=""
                type="text"
                value="B"
              />
              <label
                className="active"
                htmlFor="invType1"
              >
                Type of Investment
              </label>
            </div>
            <div
              className="input-field col s2"
            >
              <input
                className="validate"
                id="invDate1"
                name="date_1"
                onChange={[Function]}
                type="date"
                value="10-10-2001"
              />
              <label
                className="active"
                htmlFor="invDate1"
              >
                Close Date
              </label>
              <a
                href="#"
                id="deleteInvestmentButton"
                onClick={[Function]}
              >
                <i
                  className="material-icons grey-text prefix"
                >
                  delete_forever
                </i>
              </a>
            </div>
          </div>
          <a
            className="btn waves-effect waves-light light-blue"
            id="addInvestmentButton"
            onClick={[Function]}
          >
            Add Investment
          </a>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <div
          className="center"
        >
          <a
            className="modal-close waves-effect light-blue waves-light btn"
            id="submitInvestmentButton"
            onClick={[Function]}
          >
            Submit
          </a>
        </div>
      </div>
    </div>
  </div>
</AddInvestmentModal>
`;

exports[`AddInvestmentModal  renders correctly with no investments 1`] = `
<div
  className="modal"
  id="add-investments-modal"
  style={
    Object {
      "height": "75%",
      "width": "60%",
    }
  }
>
  <div>
    <div
      className="modal-content"
    >
      <h4>
        Investments for 
      </h4>
      <div
        className="center"
      >
        <a
          className="btn waves-effect waves-light light-blue"
          id="addInvestmentButton"
          onClick={[Function]}
        >
          Add Investment
        </a>
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <div
        className="center"
      >
        <a
          className="modal-close waves-effect light-blue waves-light btn"
          id="submitInvestmentButton"
          onClick={[Function]}
        >
          Submit
        </a>
      </div>
    </div>
  </div>
</div>
`;
